x-pack/legacy/plugins/siem/public/components/open_timeline/edit_timeline_batch_actions.tsx

```diff
@@ -0,0 +1,118 @@
+  dispatchToaster,
+  selectedItems,
+  tableRef,
+}: {
+  deleteTimelines: DeleteTimelines | undefined;
+  dispatchToaster: Dispatch<ActionToaster>;
+  selectedItems: OpenTimelineResult[] | undefined;
+  tableRef: React.MutableRefObject<EuiBasicTable<OpenTimelineResult> | undefined>;
+}) => {
+  const {
+    enableExportTimelineDownloader,
+    disableExportTimelineDownloader,
+    exportedIds,
+    isEnableDownloader,
+    isDeleteTimelineModalOpen,
+    onOpenDeleteTimelineModal,
+    onCloseDeleteTimelineModal,
+  }: DeleteTimeline & ExportTimeline = useEditTimelineActions(selectedItems);
+  const onCompleteBatchActions = useCallback(
+    (closePopover?: () => void) => {
+      if (closePopover != null) closePopover();
+      if (tableRef != null && tableRef.current != null) {
+        tableRef.current.changeSelection([]);
+      }
+      disableExportTimelineDownloader();
+      onCloseDeleteTimelineModal();
+    },
+    [tableRef.current]
+  );
+
+  const getBatchItemsPopoverContent = useCallback(
+    (closePopover: () => void) => {
+      return (
+        <>
+          <TimelineDownloader
+            exportedIds={exportedIds}
+            getExportedData={exportSelectedTimeline}
+            isEnableDownloader={isEnableDownloader}
+            onComplete={onCompleteBatchActions.bind(null, closePopover)}
+          />
+          {deleteTimelines != null && (
+            <DeleteTimelineModalOverlay
+              deleteTimelines={deleteTimelines}
+              onComplete={onCompleteBatchActions.bind(null, closePopover)}
+              isModalOpen={isDeleteTimelineModalOpen}
+              savedObjectIds={selectedItems?.reduce(
+                (acc, item) => (item.savedObjectId != null ? [...acc, item.savedObjectId] : acc),
+                [] as string[]
+              )}
+              title={
+                selectedItems?.length !== 1
+                  ? i18n.SELECTED_TIMELINES(selectedItems?.length ?? 0)
+                  : `"${selectedItems[0]?.title}"`
+              }
+            />
+          )}
+          <EuiContextMenuPanel
+            items={[
+              <EuiContextMenuItem
+                disabled={selectedItems == null || selectedItems.length === 0}
+                icon="exportAction"
+                key="ExportItemKey"
+                onClick={() => {
+                  enableExportTimelineDownloader();
+                }}
+              >
+                {i18n.EXPORT_SELECTED}
+              </EuiContextMenuItem>,
+              <EuiContextMenuItem
+                disabled={selectedItems == null || selectedItems.length === 0}
+                icon="trash"
+                key="DeleteItemKey"
+                onClick={() => {
+                  onOpenDeleteTimelineModal();
+                }}
+              >
+                {i18n.DELETE_SELECTED}
+              </EuiContextMenuItem>,
+            ]}
+          />
+        </>
+      );
+    },
+    [
+      deleteTimelines,
+      dispatchToaster,
```

Comment:
`dispatchToaster` and `disableExportTimelineDownloader` are not used in this callback